src/components/v2/values/chartValuesDiff/ChartValuesView.utils.ts

```diff
@@ -407,3 +201,62 @@ export const envGroupStyle = {
         transform: state.selectProps.menuIsOpen ? 'rotate(180deg)' : 'rotate(0deg)',
     }),
 }
+
+// TODO: maybe there exists an util for this in common-lib?
+const getEmptyValues = (value: any) => {
+    switch (typeof value) {
+        case 'number':
+            return 0
+        case 'string':
+            return ''
+        case 'object':
+            return Array.isArray(value) ? [] : {}
+        default:
+            return null
+    }
+}
+
+export const updateYamlDocument = (
+    operations: Operation[],
+    json: object,
+    valuesYamlDocument: YAML.Document.Parsed,
+    dispatch: React.Dispatch<ChartValuesViewAction>,
+): void => {
+    operations.forEach((operation) => {
+        const path = operation.path.slice(1).split('/')
+        if (operation.op === 'add') {
+            valuesYamlDocument.addIn(path, operation.value)
+        } else if (operation.op === 'remove') {
+            // NOTE: should never throw error since we are using the path
+            // from patch operations, it means the value should exist
+            const value = JSONPath({
+                json,
+                path: convertJSONPointerToJSONPath(operation.path),
+                resultType: 'value',
+                wrap: false,
+            })
+            // TODO: remove in production? maybe use continue statement here
+            if (!value) {
+                throw Error('failed to resolve value of remove operation path!')
+            }
+            if (typeof value === 'object') {
+                valuesYamlDocument.deleteIn(path)
+            } else {
+                valuesYamlDocument.setIn(path, getEmptyValues(value))
+            }
+        } else if (operation.op === 'replace') {
+            valuesYamlDocument.setIn(path, operation.value)
+        } else {
+            // TODO: remove in production code. Maybe handle other types of ops
```

Comment:
Should I throw an error here ?

Reply:
Please log it to sentry and suppress error

Reply:
Logging it to sentry through showError

Reply:
can use logExceptionToSentry since this is a util, we can't expect to always use showError when using this method

Reply:
this method throws error. The place where this method is used will handle this. Currently If this method fails the user won't be able to make changes through GUI mode. If we don't showError to the user it will be very frustrating for them to see that he makes changes from the GUI but they don't reflect in the actual yaml/config. This will be hard to debug, therefore just show the error to the user so that they can easily see that there is a problem and report it to us.